cmake/VASTRegisterPlugin.cmake

```diff
@@ -652,11 +652,14 @@ function (VASTRegisterPlugin)
     add_test(NAME build-${PLUGIN_TARGET}-test
              COMMAND "${CMAKE_COMMAND}" --build "${CMAKE_BINARY_DIR}" --config
                      "$<CONFIG>" --target ${PLUGIN_TARGET}-test)
-    set_tests_properties(build-${PLUGIN_TARGET}-test
-                         PROPERTIES FIXTURES_SETUP vast_unit_test_fixture)
+    set_tests_properties(
+      build-${PLUGIN_TARGET}-test
+      PROPERTIES FIXTURES_SETUP vast_${PLUGIN_TARGET}_unit_test_fixture
+                 FIXTURES_REQUIRED vast_unit_test_fixture)
```

Comment:
Does that mean that as soon as any test with `FIXTURES_SETUP vast_unit_test_fixture` completed the ones with the corresponding `FIXTURES_REQUIRED` would be unblocked?

In any case I believe a dedicated name for each plugin is much better, but why are do we need `FIXTURES_REQUIRED vast_unit_test_fixture`?

Reply:
Both the `build-${PLUGIN_TARGET}-test` and the `build-vast-test` target attempt to build `libvast`, which introduces a race between the two. This is solved by making `build-${PLUGIN_TARGET}-test` depend on `build-vast-test` with this intermediate per-plugin fixture.

Reply:
Ah, because the `build-X` tests aren't modeled in CMakes own dependency graph.

Reply:
They cannot be as they aren't targets, but rather tests, for which you can only declare dependencies via fixtures.